src/hooks/useCustomNotification.ts

```diff
@@ -0,0 +1,32 @@
+import { useCallback } from 'react';
+
+import {
+  DEFAULT_TOAST_AUTO_CLOSE_MS,
+  NotificationDisplayData,
+  NotificationType,
+} from '@/constants/notifications';
+
+import { useAppDispatch } from '@/state/appTypes';
+import { addCustomNotification } from '@/state/notifications';
+
+export const useCustomNotification = () => {
+  const dispatch = useAppDispatch();
+
+  const notify = useCallback(
+    (customNotification: Omit<NotificationDisplayData, 'groupKey'>) => {
+      dispatch(
+        addCustomNotification({
+          id: Date.now().toString(),
+          displayData: {
+            toastDuration: DEFAULT_TOAST_AUTO_CLOSE_MS,
+            groupKey: NotificationType.Custom,
```

Comment:
Do we want all custom notifications to be grouped together? I think we can keep them separate, if we want to use it for various use cases. in which case we have `groupKey: id`

Reply:
+1

Reply:
yeah that makes sense!